templates/default/systemd/alertmanager.service.erb

```diff
@@ -6,9 +6,11 @@ After=network.target
 ExecStart=<%= node['prometheus']['alertmanager']['binary'] %> \
     -log.level=debug \
     -storage.path="<%= node['prometheus']['dir'] %>/data" \
-    -config.file=/opt/prometheus/alertmanager.conf \
-    -web.external-url=http://127.0.0.1/alert-manager/
-User=prometheus
+    -config.file="<%= node['prometheus']['alertmanager']['config.file'] %>" \
+    -web.external-url="<%= node['prometheus']['flags']['alertmanager.url'] %>"
```

Comment:
EC - you might want to add a test to verify that you did all of the attribute setting correctly and ended up with the value in the template that you expected.